### PR TITLE
Add postgres JDBC source

### PR DESCRIPTION
**Pull request by sherifnada**
## What
* Adds the Postgres JDBC-based source. 
* Bumps postgres dependency to the latest version across the repo

## Reading order
1. `PostgresSource.java`
1. `PostgresSourceTest.java`
1. `PostgresIntegrationTest.java`
1. everything else

## Checklist
- [x] publish v0.1.0 once this PR is ready and use it in Core

### CODE REVIEW

File: airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresSourceTest.java [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  private static final AirbyteCatalog CATALOG = CatalogHelpers.createAirbyteCatalog(
+      STREAM_NAME,
+      Field.of("id", JsonSchemaPrimitive.NUMBER),
+      Field.of("name", JsonSchemaPrimitive.STRING));
+  private static final Set<AirbyteMessage> ASCII_MESSAGES = Sets.newHashSet(
+      new AirbyteMessage().withType(Type.RECORD)
+          .withRecord(new AirbyteRecordMessage().withStream(STREAM_NAME).withData(Jsons.jsonNode(ImmutableMap.of("id", 1, "name", "goku")))),
+      new AirbyteMessage().withType(Type.RECORD)
+          .withRecord(new AirbyteRecordMessage().withStream(STREAM_NAME).withData(Jsons.jsonNode(ImmutableMap.of("id", 2, "name", "vegeta")))),
+      new AirbyteMessage().withType(Type.RECORD)
```
Comment (michel-tricot) [inline airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresSourceTest.java:77]: Forgetting beeru

**Review by michel-tricot — APPROVED** [review]

File: airbyte-integrations/connectors/source-postgres/build.gradle [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    implementation project(':airbyte-db')
+    implementation project(':airbyte-integrations:bases:base-java')
+    implementation project(':airbyte-protocol:models')
+    implementation project(':airbyte-integrations:connectors:source-jdbc')
+
+    testImplementation 'org.apache.commons:commons-text:1.9'
+    testImplementation 'org.apache.commons:commons-lang3:3.11'
+    testImplementation 'org.apache.commons:commons-dbcp2:2.7.0'
+    testImplementation 'com.fasterxml.jackson.core:jackson-databind'
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-postgres/build.gradle:19]: why is this needed? isn't it imported to all of our java gradle projects automatically?

**Review by cgardens — APPROVED** [review]
